test/fw/ptl/utils/pbs_crayutils.py

```diff
@@ -98,8 +103,16 @@ def parse_apstat(self, options='-rn'):
                 break
             index = 0
             obj = {}
-            for val in line.split():
-                obj[keys[index]] = val
+            line = line.split()
+            for i in range(0, len(line)):
+                obj[keys[index]] = line[i]
+                if keys[index] == 'State':
+                    obj[keys[index]] = line[i] + "  " + line[i + 1]
+                index += 1
+            if "Apids" in obj:
+                pass
+            else:
+                obj["Apids"] = None
```

Comment:
This can be refined to 'if "Apids' not in obj:' without the else case.

Reply:
I wanted to use 'None' as the value if there is no Apids value shown in apstat.